### ISSUE TITLE
Homework 1 Report

### ISSUE THREAD

**Issue by dzakich** [DLWM-MSE/CPSC-544#7]
Gents, the report framework is complete. We can and should start putting down all the information that is currently available and confirmed. 
For instance, we can freely work on sections about **Work Products, Roles, Practices** , all sections within **Additional Activities**, some **Pre-Game** sections, etc.

It is good to initiate the creation of content; as long as we have something, we can see how each of us think about different parts of this process; this way we can iteratively refine it based on our communal input. Also, progressively working on a content we mitigate the risk of finding ourselves having gaps in a shared vision of what we are trying to accomplish 🍭

**Comment by dzakich** [comment]
[https://docs.google.com/document/d/1T9HDjBW_mW6BGpAzWxFLnG5gfyZCHkLeRD5CTdsbRS8/edit#heading=h.x1ujium4mmkv](url)

**Comment by dzakich** [comment]
Additional Details:
1. GoogleDrive doc is version controlled. All context changes are automatically saved by Google cloud services.
2. After making the addition/alternation to context anywhere within the document, please ensure to add a revision change not to the corresponding section of the document. 
3. The style of the headers and format is defined and needs to be observed. _Heading1_, _Heading2_, _Heading3_, and _Heading4_ are defined and used. Body font is **Georgia 12 double-spaced**. 
All to-be-removed contextual clues and in-line comments are designated by
**<** _comment here_ **>** brackets. 
4. Suggest context changes using built-in functionality of __Comments__ or __Suggest Edits__ using Right-Click context menu on selected section

**Comment by jlarceo40** [comment]
"After making the addition/alternation to context anywhere within the document, please ensure to add a revision change *not to* the corresponding section of the document." Did you mean "please ensure to add a revision change *to* the corresponding section of the document." ?
Also, do we want to version any changes to the document? If we do, the revision change page could get unnecessarily large.

How did we determine that the document is 12 and double-spaced? Not against the idea, I just want to make sure that the professor will allow that. Did I miss reading that somewhere?

Work update: I will go ahead and cover the Work Products, Roles, and Practices section. I have finished the first draft for the User Story, and I have written something down for CRC Cards. I have added another Work Product: Class Diagrams. I have written something short for that too, for now.

**Comment by jlarceo40** [comment]
Also, the Google Drive link posted does not work. I just went into Google Drive to access the doc.

**Comment by dzakich** [comment]
1. Yes, is was a typo. ...add a revision change __to__ the corresponding section of the document.

2. Document revision and revision table is mandated by the homework. To prevent the revision history from growing out-of-bounds, we can only document major additions/revision of the content.

3. It seems that __12 double-spaced__ has always been a default format for all the assignments I can recall doing. Does not hurt to ask the professor though. Even though these settings will surely alter the physical length of any section, we should not be too concerned about page limits established by the prof; right now I feel like we would certainly exceed the minimum limits for each of the sections.

4. Work Products, Roles, and Practices is a pretty big section. For each of the entries there will be a solid block of information. Since there are multiple Work PRoducts, Roles, and Practices it might be too much of a work for a single person, Jonny. If you decided to work on it, I'd recommend putting good effort in 1 Work Product, 1 Role, and 1 Practice to serve as an example for all of us who will work on the rest.

**Comment by dzakich** [comment]
Oh yea, Eric and I noticed that sometimes GitHub seems to break the embedded links.

**Comment by jlarceo40** [comment]
True, that section will be huge but I am on a roll in terms of things I can write down for Work Products and Roles; I did a lot of reading this weekend and therefore, I have a lot of information to put down. I am confident that I cover plenty on those topics. Basically, I'll put down what I confidently know, then we can go from there.

**Comment by dzakich** [comment]
@jlarceo40 🏆

**Comment by dzakich** [comment]
@DLWM-MSE/alps 
Gents, I am working on __Vision__ section of the document. Just completed __Vision : Application Description__ section.
Would love to receive your feedback on it. Currently working on Project goals, org goals, sys features, an context diagram.

**Comment by bengk** [comment]
Here are some functional requirements that I came up with for the rest of the user stories. I didn't want to just put this in without a discussion so I'll post it here since it's HW1 related. Keep? Change? Remove? Discuss. 



 

2e. As a user I want the software to notify me when the sorting is completed so that I can continue or quit the application. 

- The software will ask the user with a descriptive question on whether the user wants to continue, while giving the user the proper reply necessary to continue or quit. 

- The software will continue to show the proper descriptive instructions on the required inputs to continue another sorting iteration if the user decides to continue. 

2f. As a user I want to know all valid controls so that I don’t have to guess them and properly operate the whole software.

- Depending on the prompt from the software it will include the primary controls necessary to continue at the end of each prompt in a concise manner. 

3a. As a user I want the software to detect improper commands so that I can re-enter the proper commands. 

- The software will respond to an incorrect input with a list of the correct inputs that is required to continue. 

4a. As a user I want to be able to turn off the intermediary step outputs so that I can immediately get the results.

- The software will have the option to continue through the sorting without prompting the user for any actions until the very end. This option can be applied at the beginning or any of the intermediate sorting iterations with a given input from the user.
